### PR TITLE
Remove extra log PVC

### LINKED ISSUE #387: 0.4 release must not try to create multiple PVCs across controllers
## Expected Behavior
Folks who try to use Pipelines after our 0.4 release should not have to work around the issue we found in #375, which effectively makes it infeasible to use Pipelines with regional GKE clusters.

## Actual Behavior
If you try to use Pipelines with a GKE regional cluster, and you try to use a Task with an output, the resulting pod for that Task will be unschedulable.

This is because two PVCs will be created, one by the PipelineRun controller for the output resource, and one by the TaskRun controller for the logs. These will be assigned round robin to different zones within the region, and then the resulting pod will be unable to be scheduled such that it can use both (it can't be in 2 zones at once).

## Additional Info
We have several options for how to resolve this before we release:
1. If we finish #224 before the 0.4 release, we can remove the log PVC because users should be able to reliably (enough) get logs from the step containers themselves
2. If we complete #384, we could potentially get rid of the output PVC entirely
3. If we neither of the above complete, we should remove the PVC we added only for logs

It is arguable that (2) is not enough, and we should remove the extra log PVC regardless, since we shouldn't be leaking PVCs. The main benefit of the log PVC as it stands at the moment is that it allows us to check log output in end to end tests.

### PR DESCRIPTION
**Pull request by bobcatfish**
We noticed early on that logs from init containers are often cleaned up
immediately by k8s, particularly if the containers are short running
(e.g. just echoing "hello world"). We started down a path to correct
that, which takes an approach based on Prow's entrypoint solution
(https://github.com/kubernetes/test-infra/tree/master/prow/cmd/entrypoint)
(even using the same image at the moment!) which wraps the user's
provided command and streams logs to a volume, from which the logs can
be uploaded/streamed by a sidecar.

Since we are using init containers for step execution, we can't yet use
sidecars, but we are addressing that in #224 (also an entrypoint
re-writing based solution). Once we have that, we can sidecar support,
starting with GCS as a POC (#107) and moving into other types.

In the meantime, to enable us to get logs (particularly in tests), we
had the taskrun controller create a PVC on the fly to hold logs. This
has two problems:
* The PVCs are not cleaned up so this is an unexpected side effect for
  users
* Combined with PVC based input + ouput linking, this causes scheduling
  problems for the resulting pods (#375)

Now that we want to have an official release, this would be a bad state
to release in, so we will remove this magical log PVC creation logic,
which was never our intended end state anyway.

Since we _do_ need the entrypoint rewriting and log interception logic
in the long run, this commit leaves most functionality intact, removing
only the PVC creation and changing the volume being used to an
`emptyDir`, which is what we will likely use for #107 (and this is how
Prow handles this as well). This means the released functionality will
be streaming logs to a location where nothing can read them, however I
think it is better than completely removing the functionality b/c:
1. We need the functionality in the long run
2. Users should be prepared for this functionality (e.g. dealing with
   edge cases around the taskrun controller being able to fetch an
   image's entrypoint)

Fixes #387

(@pivotal-nader-ziada I think you said you've been refactoring the PVC logic a bunch, so if this conflicts with your changes plz just ignore it, and I'll deal with any rebasing later after you've merged!)

### CODE REVIEW

**Comment by bobcatfish** [comment]
```
I0127 20:35:41.594]  clusterConditions: []
I0127 20:35:41.595]  detail: u"Deploy error: Not all instances running in IGM after 4m17.390316985s. Expect 1. Current errors: [ZONE_RESOURCE_POOL_EXHAUSTED_WITH_DETAILS]: Instance 'gke-kbuild-pipeline-e2e--default-pool-11f3aa8c-vsgj' creation failed: The zone 'projects/knative-boskos-05/zones/us-central1-f' does not have enough resources available to fulfill the request.  '(resource type:pd-standard)'. - ; ."
```

🤔 

/test pull-knative-build-pipeline-integration-tests

**Comment by bobcatfish** [comment]
Looks like something is wrong with the cluster setup :S gonna have to ping @adrcunha and folks once we're back in regular working hours.

**Comment by pivotal-nader-ziada** [comment]
@bobcatfish this PR just needs to resolve conflicts and would be good to get merged

**Comment by bobcatfish** [comment]
```
I0130 22:04:32.662] # github.com/knative/build-pipeline/test
I0130 22:04:32.663] test/embed.go:68:18: undefined: setup
I0130 22:04:32.663] test/embed.go:70:42: undefined: tearDown
I0130 22:04:32.663] test/embed.go:71:8: undefined: tearDown
```

what have i done

File: docs/using.md [hunk 1]
```diff
@@ -123,9 +123,14 @@ specific contract.
 #### Entrypoint
 
 When containers are run in a `Task`, the `entrypoint` of the container will be
-overwritten with a custom binary that redirects the logs to a separate location
-for aggregating the log output. As such, it is always recommended to explicitly
-specify a command.
+overwritten with a custom binary. The plan is to use this custom binary for
+controlling the execution of step containers ([#224](https://github.com/knative/build-pipeline/issues/224)) and log streaming
+[#107](https://github.com/knative/build-pipeline/issues/107), though currently
+it will write logs only to an [`emptyDir`]() (which cannot be read from after
```
Comment (bobcatfish) [inline docs/using.md]: whoops 😅

**Pull request merged**